Numbers/StaticConstructorOnGameStart.cs

```diff
@@ -90,9 +92,34 @@ private static void PopulatePresets()
             combatPreset.AddRange(DefDatabase<PawnTableDef>.GetNamed("Numbers_CombatPreset").columns);
             workTabPlusPreset.AddRange(DefDatabase<PawnTableDef>.GetNamed("Numbers_WorkTabPlusPreset").columns);
             colonistNeedsPreset.AddRange(DefDatabase<PawnTableDef>.GetNamed("Numbers_ColonistNeedsPreset").columns);
+            PopulatePsycastingPreset();
             PopulateMedicalPreset();
         }
 
+        private static void PopulatePsycastingPreset()
+        {
+            psycastingPreset.Add(DefDatabase<PawnColumnDef>.GetNamed("Label"));
+            psycastingPreset.Add(DefDatabase<PawnColumnDef>.GetNamed("Numbers_PsylinkLevel"));
+            psycastingPreset.Add(DefDatabase<PawnColumnDef>.GetNamed("Numbers_Psyfocus"));
+            psycastingPreset.Add(DefDatabase<PawnColumnDef>.GetNamed("Numbers_Entropy"));
+            List<PawnColumnDef> abilities = DefDatabase<PawnColumnDef>.AllDefsListForReading.Where(pcd => pcd.Ext()?.ability != null).ToList();
+            abilities.Sort(delegate (PawnColumnDef xpcd, PawnColumnDef ypcd)
+            {
+                AbilityDef x = xpcd.Ext().ability;
+                AbilityDef y = ypcd.Ext().ability;
+                int levelCompare = x.level.CompareTo(y.level);
+                int psyfocusCompare = x.PsyfocusCost.CompareTo(y.PsyfocusCost);
+                int entropyCompare = x.EntropyGain.CompareTo(y.EntropyGain);
+                if (levelCompare != 0) { return levelCompare; }
+                if (psyfocusCompare != 0) { return psyfocusCompare; }
+                if (entropyCompare != 0) { return entropyCompare; }
```

Comment:
This is awful! I don't know how to do cascading tiebreak comparators in C# delegates!

Reply:
I figured some sorting was appropriate. It seems appropriate to me. With this they are ordered left-to-right (low to high) first by Level, then by Psyfocus Cost, then by Neural Heat cost, which actually gives them a pretty intuitive grouping, where they are clustered by level, with combat abilities tending to the left (due to low psyfocus cost), and the non-combat 1.2 abilities tending to the right (with very high psyfocus costs).

The sorting by level, I think, is pretty intuitive, as a pawn gets higher level psylinks, they'll start to fill squares roughly left-to-right, with the most difficult (high level) abilities at the far right. It kinda naturally mimics progression.

Reply:
![sortingResults](https://user-images.githubusercontent.com/7416299/101961947-dce67e80-3bd8-11eb-98d6-1bd6347869af.png)

Reply:
I left the sorting for now (per above) but I made the code better and terser with some LINQ-fu.